plenum/test/req_drop/test_req_drop_on_propagate_phase.py

```diff
@@ -0,0 +1,98 @@
+import pytest
+
+from plenum.common.constants import PROPAGATE
+from plenum.test.helper import sdk_json_to_request_object, sdk_send_random_requests
+from stp_core.loop.eventually import eventually
+from plenum.common.messages.node_messages import Propagate
+from plenum.test.delayers import delay, msg_rep_delay
+from plenum.test.propagate.helper import recvdRequest, recvdPropagate, \
+    sentPropagate
+from plenum.test.test_node import TestNode
+from plenum.test.node_request.helper import sdk_ensure_pool_functional
+
+
+howlong = 20
+initial_ledger_size = 0
+
+
+@pytest.fixture(scope="module")
+def tconf(tconf):
+    OUTDATED_REQS_CHECK_ENABLED_OLD = tconf.OUTDATED_REQS_CHECK_ENABLED
+    OUTDATED_REQS_CHECK_INTERVAL_OLD = tconf.OUTDATED_REQS_CHECK_INTERVAL
+    PROPAGATES_PHASE_REQ_TIMEOUT_OLD = tconf.PROPAGATES_PHASE_REQ_TIMEOUT
+    ORDERING_PHASE_REQ_TIMEOUT_OLD = tconf.ORDERING_PHASE_REQ_TIMEOUT
+
+    tconf.OUTDATED_REQS_CHECK_ENABLED = True
+    tconf.OUTDATED_REQS_CHECK_INTERVAL = 1
+    tconf.PROPAGATES_PHASE_REQ_TIMEOUT = 3
+    tconf.ORDERING_PHASE_REQ_TIMEOUT = 3600
+    yield tconf
+
+    tconf.OUTDATED_REQS_CHECK_ENABLED = OUTDATED_REQS_CHECK_ENABLED_OLD
+    tconf.OUTDATED_REQS_CHECK_INTERVAL = OUTDATED_REQS_CHECK_INTERVAL_OLD
+    tconf.PROPAGATES_PHASE_REQ_TIMEOUT = PROPAGATES_PHASE_REQ_TIMEOUT_OLD
+    tconf.ORDERING_PHASE_REQ_TIMEOUT = ORDERING_PHASE_REQ_TIMEOUT_OLD
+
+
+@pytest.fixture()
+def setup(txnPoolNodeSet, looper, sdk_pool_handle, sdk_wallet_client):
+    global initial_ledger_size
+    A, B, C, D = txnPoolNodeSet  # type: TestNode
+    delay(Propagate, frm=[B, C, D], to=A, howlong=howlong)
+    # Delay MessageRep by long simulating loss as if Propagate is missing, it
+    # is requested
+    A.nodeIbStasher.delay(msg_rep_delay(10 * howlong, [PROPAGATE, ]))
+    initial_ledger_size = txnPoolNodeSet[0].domainLedger.size
+    request_couple_json = sdk_send_random_requests(
+        looper, sdk_pool_handle, sdk_wallet_client, 1)
+    return request_couple_json
+
+
+def testReqDropOnPropagatePhaseOnMasterPrimaryAndThenOrdered(tconf, setup, looper, txnPoolNodeSet,
```

Comment:
What other tests we are going to add here? I would like to see the list.

Reply:
Will list in the JIRA ticket INDY-1836.

Reply:
Will list in the JIRA ticket INDY-1836.